mullvad-relay-selector/tests/relay_selector.rs

```diff
@@ -706,6 +716,242 @@ fn test_selecting_any_relay_will_consider_multihop() {
     }
 }
 
+/// Test whether Shadowsocks is always selected as the obfuscation protocol when Shadowsocks is
+/// selected.
+#[test]
+fn test_selecting_wireguard_over_shadowsocks() {
+    let RelayListWithExtraShadowsocksIps { relay_list, .. } =
+        get_relay_list_with_extra_shadowsocks_addresses();
+
+    let relay_selector = RelaySelector::from_list(SelectorConfig::default(), relay_list);
+
+    let mut query = RelayQueryBuilder::new().wireguard().shadowsocks().build();
+    query.wireguard_constraints.use_multihop = Constraint::Only(false);
+
+    let relay = relay_selector.get_relay_by_query(query).unwrap();
+    match relay {
+        GetRelay::Wireguard {
+            obfuscator,
+            inner: WireguardConfig::Singlehop { .. },
+            ..
+        } => {
+            assert!(obfuscator.is_some_and(|obfuscator| matches!(
+                obfuscator.config,
+                ObfuscatorConfig::Shadowsocks { .. }
+            )))
+        }
+        wrong_relay => panic!(
+            "Relay selector should have picked a Wireguard relay with Shadowsocks, instead chose {wrong_relay:?}"
+        ),
+    }
+}
+
+/// Test whether extra Shadowsocks IPs are selected when available
+#[test]
+fn test_selecting_wireguard_over_shadowsocks_extra_ips() {
+    let RelayListWithExtraShadowsocksIps {
+        relay_list,
+        shadowsocks_extra_relay,
+        shadowsocks_extra_addr_in,
+    } = get_relay_list_with_extra_shadowsocks_addresses();
+
+    let relay_selector = RelaySelector::from_list(SelectorConfig::default(), relay_list);
+
+    let mut query = RelayQueryBuilder::new().wireguard().shadowsocks().build();
+    query.wireguard_constraints.use_multihop = Constraint::Only(false);
```

Comment:
Perhaps this should be expressable using the builder? E.g. `RelayQueryBuilder::new().wireguard().multihop(false).shadowsocks().build()`. What do you think?